src/mito2/src/read/scan_util.rs

```diff
@@ -14,23 +14,301 @@
 use futures::Stream;
 use prometheus::IntGauge;
 use store_api::storage::RegionId;
 
 use crate::error::Result;
-use crate::metrics::IN_PROGRESS_SCAN;
+use crate::metrics::{
+    IN_PROGRESS_SCAN, PRECISE_FILTER_ROWS_TOTAL, READ_BATCHES_RETURN, READ_ROWS_IN_ROW_GROUP_TOTAL,
+    READ_ROWS_RETURN, READ_ROW_GROUPS_TOTAL, READ_STAGE_ELAPSED,
+};
 use crate::read::range::{RangeBuilderList, RowGroupIndex};
 use crate::read::scan_region::StreamContext;
 use crate::read::{Batch, ScannerMetrics, Source};
 use crate::sst::file::FileTimeRange;
 use crate::sst::parquet::reader::ReaderMetrics;
 
+/// Scan metrics to report to the execution metrics.
+struct ScanMetricsSet {
+    /// Duration to prepare the scan task.
+    prepare_scan_cost: Time,
+    /// Duration to build the (merge) reader.
+    build_reader_cost: Time,
+    /// Duration to scan data.
+    scan_cost: Time,
+    /// Duration to convert batches.
+    convert_cost: Time,
+    /// Duration while waiting for `yield`.
+    yield_cost: Time,
+    /// Duration of the scan.
+    total_cost: Time,
+    /// Number of rows returned.
+    num_rows: Count,
+    /// Number of batches returned.
+    num_batches: Count,
+    /// Number of mem ranges scanned.
+    num_mem_ranges: Count,
+    /// Number of file ranges scanned.
+    num_file_ranges: Count,
+
+    // SST related metrics:
+    /// Duration to build file ranges.
+    build_parts_cost: Time,
+    /// Number of row groups before filtering.
+    rg_total: Count,
+    /// Number of row groups filtered by fulltext index.
+    rg_fulltext_filtered: Count,
+    /// Number of row groups filtered by inverted index.
+    rg_inverted_filtered: Count,
+    /// Number of row groups filtered by min-max index.
+    rg_minmax_filtered: Count,
+    /// Number of row groups filtered by bloom filter index.
+    rg_bloom_filtered: Count,
+    /// Number of rows in row group before filtering.
+    rows_before_filter: Count,
+    /// Number of rows in row group filtered by fulltext index.
+    rows_fulltext_filtered: Count,
+    /// Number of rows in row group filtered by inverted index.
+    rows_inverted_filtered: Count,
+    /// Number of rows in row group filtered by bloom filter index.
+    rows_bloom_filtered: Count,
+    /// Number of rows filtered by precise filter.
+    rows_precise_filtered: Count,
+    /// Number of record batches read from SST.
+    num_sst_record_batches: Count,
+    /// Number of batches decoded from SST.
+    num_sst_batches: Count,
+    /// Number of rows read from SST.
+    num_sst_rows: Count,
+
+    /// Elapsed time before the first poll operation.
+    first_poll: Time,
+}
+
+impl std::fmt::Debug for ScanMetricsSet {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("ScanMetricsSet")
+            .field(
+                "prepare_scan_cost",
+                &Duration::from_nanos(self.prepare_scan_cost.value() as u64),
```

Comment:
Suggest to use `human-time` crate to format the raw time cost here, better for people to see.

Reply:
Debug print will format the duration in human-readable form.